### PR TITLE
Command line interface

### LINKED ISSUE #6: Command line interface
**Is your feature request related to a problem? Please describe.**

Running this code always through a Python session or a notebook is overly slow, a command line interface would super nice.

**Describe the solution you'd like**

Use the packages described in [this blog post](https://stormpath.com/blog/building-simple-cli-interfaces-in-python) and also add in a few other tricks they recommend (making sure to update docs as well as CI etc. when doing this).

**Describe alternatives you've considered**

No command line options...

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Add a basic command line interface

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <http://link-to-pr.com>`_) Added feature which does something
- (`#XX <http://link-to-pr.com>`_) Fixed bug identified in (`#XX <http://link-to-issue.com>`_)
```

closes #6

### CODE REVIEW

**Comment by znicholls** [comment]
@lewisjared this can go in 0.6.1 so the crunching can start running (I've done a 0.6.1-alpha2 pre-release)

File: notebooks/crunch-cmip5-data.ipynb [hunk 1]
```diff
@@ -18,30 +18,35 @@
...
-      "                            input output\r\n",
+      "Usage: netcdf-scm-crunch [OPTIONS] SRC DST\r\n",
       "\r\n",
-      "Crunch netCDF files to NetCDF SCM csv files\r\n",
+      "  Crunch data in ``src`` to OpenSCM csv's in ``dst``.\r\n",
       "\r\n",
-      "positional arguments:\r\n",
-      "  input                 Root folder of the data to crunch.\r\n",
-      "  output                Root folder in which to save the output data\r\n",
+      "  ``src`` is searched recursively and netcdf-scm will attemp to crunch all\r\n",
```
Comment (lewisjared) [inline notebooks/crunch-cmip5-data.ipynb]:
```suggestion
      "  ``src`` is searched recursively and netcdf-scm will attempt to crunch all\r\n",
```

File: setup.py [hunk 2]
```diff
@@ -5,48 +5,76 @@
...
+    "Intended Audience :: Developers",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3.7",
+]
 
-DESCRIPTION = (
-    "Python wrapper for processing netCDF files for use with simple climate models"
-)
-README = "README.rst"
+ENTRY_POINTS = {"console_scripts": ["netcdf-scm-crunch = netcdf_scm.cli:crunch_data"]}
```
Comment (lewisjared) [inline setup.py]: Why not just netcdf-scm? Are you planning on having multiple CLI tools?

File: tests/integration/test_crunching.py [hunk 3]
```diff
@@ -1,40 +1,48 @@
...
-        INPUT_DIR,
-        OUTPUT_DIR,
-        "--var-to-crunch",
-        VAR_TO_CRUNCH,
-        "-f",
-    ]
-    subprocess.check_call(command)
+    VAR_TO_CRUNCH = ".*tas.*"
+
+    runner = CliRunner()
```
Comment (lewisjared) [inline tests/integration/test_crunching.py]: Nice! I always struggled to test argparse cli's

File: setup.py (hunk 2 above)
Comment (znicholls) [reply setup.py]: yes

File: tests/integration/test_crunching.py (hunk 3 above)
Comment (znicholls) [reply tests/integration/test_crunching.py]: Click is so nice

**Pull request merged**